File: backend/src/main/java/helper/RandomStringGenerate.java
package helper;

import java.util.Locale;
import java.util.Random;

//Quelle: https://stackoverflow.com/questions/41107/how-to-generate-a-random-alpha-numeric-string
public class RandomStringGenerate {

    public static final String upper = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";

    public static final String lower = upper.toLowerCase(Locale.ROOT);

    public static final String digits = "0123456789";

    public static final String alphanum = upper + lower + digits;

    private final Random random = new Random();
    private final char[] symbols = alphanum.toCharArray();

    /**
     * Generate a random string.
     */
    public String GenerateRandomString(int length) {
        char[] buf = new char[length];

        for (int idx = 0; idx < buf.length; ++idx)
            buf[idx] = symbols[random.nextInt(symbols.length)];
        return new String(buf);
    }
}

File: Backend/Core/Helper/RandomStringGenerate.cs
namespace Core.Helper;

public class RandomStringGenerate
{
    private static readonly string Upper = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
    private static readonly string Lower = Upper.ToLower();
    private static readonly string Digits = "0123456789";
    private static readonly string Alphanum = Upper + Lower + Digits;

    private readonly Random _random = new Random();
    private readonly char[] _symbols = Alphanum.ToCharArray();

    /// <summary>
    /// Generates a random alphanumeric string of the specified length.
    /// </summary>
    /// <param name="length">The length of the string to generate.</param>
    /// <returns>A random alphanumeric string.</returns>
    public string GenerateRandomString(int length)
    {
        var buffer = new char[length];

        for (int idx = 0; idx < buffer.Length; ++idx)
        {
            buffer[idx] = _symbols[_random.Next(_symbols.Length)];
        }

        return new string(buffer);
    }
}